sockeye/arguments.py

```diff
@@ -175,7 +175,32 @@ def add_model_parameters(params):
                               choices=C.ENCODERS,
                               default=C.RNN_NAME,
                               help="Type of encoder. Default: %(default)s.")
+    model_params.add_argument('--decoder',
+                              choices=C.DECODERS,
+                              default=C.RNN_NAME,
+                              help="Type of encoder. Default: %(default)s.")
+
+    model_params.add_argument('--num-layers',
```

Comment:
Not a bad idea. We could extend that to the names for encoder/decoder flags as well, e.g. --model rnn:transformer or transformer.
Equally for num_embed: 512:1024, vocab-size etc etc.
I think this would be a nice change but out of scope of this PR.
I will leave as is for now.